src/inspect.rs

```diff
@@ -189,3 +209,106 @@ impl MetadataPrinter for MetadataPrettyPrinter {
         self.print_metadata_usage(metadata)
     }
 }
+
+fn get_signatures_printer(output_type: OutputType) -> Box<dyn SignaturesPrinter> {
+    match output_type {
+        OutputType::Yaml => Box::new(SignaturesYamlPrinter {}),
+        OutputType::Pretty => Box::new(SignaturesPrettyPrinter {}),
+    }
+}
+
+trait SignaturesPrinter {
+    fn print(&self, signatures: &OciImageManifest);
+}
+
+struct SignaturesPrettyPrinter {}
+
+impl SignaturesPrinter for SignaturesPrettyPrinter {
+    fn print(&self, signatures: &OciImageManifest) {
+        for layer in &signatures.layers {
+            let mut table = Table::new();
+            table.set_format(FormatBuilder::new().padding(0, 1).build());
+            table.add_row(row![Fmbl -> "Digest: ", layer.digest]);
+            table.add_row(row![Fmbl -> "Media type: ", layer.media_type]);
+            table.add_row(row![Fmbl -> "Size: ", layer.size]);
+            if let Some(annotations) = &layer.annotations {
+                table.add_row(row![Fmbl -> "Annotations"]);
+                for annotation in annotations.iter() {
+                    table.add_row(row![Fgbl -> annotation.0, annotation.1]);
+                }
+            }
+            table.printstd();
+            println!();
+        }
+    }
+}
+
+struct SignaturesYamlPrinter {}
+
+impl SignaturesPrinter for SignaturesYamlPrinter {
+    fn print(&self, signatures: &OciImageManifest) {
+        let signatures_yaml = serde_yaml::to_string(signatures);
+        if let Ok(signatures_yaml) = signatures_yaml {
+            println!("{}", signatures_yaml)
+        }
+    }
+}
+
+async fn fetch_signatures_manifest(
+    uri: &str,
+    sources: Option<Sources>,
+    docker_config: Option<DockerConfig>,
+) -> Option<OciImageManifest> {
```

Comment:
I think it would be better to change the return value of  this function to be `Result<Option<OciImageManifest>>`.

The current code swallows errors and turn them into `None`. That can lead the user to assume a policy is not signed, while instead there's just a network hiccup that prevents the code from fetching the signatures.

Note well: I don't think we should exit with an error as soon as the signatures cannot be fetched. We should still print all the information about the policy and then, inside of the signature section, we should write something like "Cannot determine if the policy has been signed. There was an error while attempting to fetch its signatures from the remote registry: <error>".